sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/CLIService.java

```diff
@@ -126,7 +126,7 @@ private void applyAuthorizationConfigPolicy(HiveConf newHiveConf) throws HiveExc
     // authorization and authentication are not session specific settings
     SessionState ss = new SessionState(newHiveConf);
     ss.setIsHiveServerQuery(true);
-    SessionState.start(ss);
```

Comment:
Seems there is a bug originally here as the session state is not closed and therefore will leave the session directories orphan.

Reply:
Revert this change because Spark use `SparkSQLCLIService`.